### DISCUSSION TITLE
How to add new annotations on which we can filter?

### DISCUSSION THREAD

**Discussion by jxchong** [broadinstitute/seqr#2174] [Q&A]
How do we specify new/additional annotations on which we can filter and then view the value in the variant search result interface? 

Some annotations we need to load/use will be values embedded in a tag in the INFO field of a sites-only VCF (example use cases include frequency from our internal control database, beta versions of CADD or other pathogenicity predictors) and others are in a BED file with the last column containing the annotation/label (e.g. a BED file containing tissue-specific enhancers or a file flagging all "OMIM genes")

**Comment by hanars** [comment]
To clarify, are you asking how to update an open source deployment of seqr to support additional annotations, or are you asking to update the core seqr code base?

**Reply by jxchong** [reply]
Mostly the former / is it possible currently? Or is this a feature that would need development?

**Comment by hanars** [comment]
seqr does not support installation-specific data, nor do we intend to. If you want to add additional annotations to seqr in your own installation, you need to update the seqr codebases in a several places:
1) Update the loading pipeline code to pull and export additional annotations. This should be possible for annotations anywhere within the VCF, including the INFO fields. Adding an additional file to the loading pipeline, however, would be a very large change to the whole flow of the pipeline, and I would question its value. Gene level annotations should never be added at the pipeline stage, only variant-level annotations. For example, seqr already flags and allows you to sort on OMIM genes, and these were very intentionally not added to the pipeline but rather to the reference data stored directly within seqr. If you are trying to add site-specific annotations not available in the VCF itself, your best bet would be to take the publicly-available seqr reference data matrix table, create your own copy, add any of your own annotations to that MT using hail, and then update the pipeline to use your reference MT instead of the default. However, that does mean that if the seqr team ever provides updated reference data you would be unable to use it without substantial effort on your end. 
2) If needed, update the seqr search page UI to add additional filters for the new fields. Then update the backend search code to parse the filters sent from the front end and transform them into a valid elasticsearch query on the new fields
3) Update the elasticsearch code for parsing data from elasticsearch to return any new fields from elasticsearch and return them as part of search results. Then update the UI for variant display to show those fields

**Reply by jxchong** [reply]
Ok, and if we wanted to nominate something for inclusion in core seqr, I assume we just have to create a ticket and ask for consideration?

**Reply by hanars** [reply]
yes exactly. Its worth noting though that due to the complexity of adding annotations even if we do agree to include something it may be a while before we have the bandwidth to add it
